Update maia after introduction of long-short terminology

luckysori opened the pull request:
Fixes #1580.
Fixes #1482.

Blocked by https://github.com/comit-network/maia/issues/16. (We have to amend the last commit and depend on latest `maia` after https://github.com/comit-network/maia/pull/21 is merged).

With this change we also fix the fact that we were (wrongly) only using the `calculate_payouts_long_taker` API. With the new semantics, we can always call the generic `calculate_payouts` API since it accepts arguments based on long/short.

The main drawback of this change is that we've elected to introduce the `Position` into the `Dlc` which is now duplicated across `Cfd` and `Dlc`. This may not be the final solution, but it helps in being able to use ours/theirs and long/short semantics with the data inside of `Dlc`.

The hardest pill to swallow might be the changes to the tests in `cfd.rs`, where a few test APIs have been modified to be able to pass through the `Position`.

luckysori committed:
Clarify documentation of SETTLEMENT_INTERVAL

As a side-effect it no longer mentions maker and taker.

luckysori committed:
Update OpeningFee doc-string

luckysori committed:
Rename slightly confusing tests

luckysori committed:
Update EventKind doc-string

thomaseizinger commented on model/src/cfd.rs:
```diff
@@ -960,8 +960,8 @@ impl Cfd {
-            taker: *payout.taker_amount(),
-            maker: *payout.maker_amount(),
+            taker: *payout.long_amount(),
+            maker: *payout.short_amount(),
```
Well, this is not going to be accurate now is it? :D

luckysori replied:
Hehe it looks wrong, but it is in the context of having called:

https://github.com/itchysats/itchysats/pull/1577/files/e75b61cdf4f0857a389e67e00c415d97bd54615d#diff-f078e5601c18fcb29f773d04d16902354c8f66a57b41aedaea122e11a6f8fd3dR943-R950

So it would actually work here.

In any case, I'm still working on this PR because there are other mistakes.

luckysori committed:
Check role when performing role-specific methods on Cfd

luckysori committed:
Add Position to Dlc struct

It will help after we transition away from having knowledge of the
maker and taker roles inside of the `Dlc`.

luckysori committed:
Remove unused method on Dlc

luckysori committed:
Comment on reason for using arbitrary script pubkey

luckysori committed:
Use long/short terminology around model::Cfd

With this change we also fix the fact that we were (wrongly) only
using the `calculate_payouts_long_taker` API. With the new semantics,
we can always call the generic `calculate_payouts` API since it
accepts arguments based on long/short.

The main drawback of this change is that we've elected to introduce
the `Position` into the `Dlc` which is now duplicated across `Cfd` and
`Dlc`. This may not be the final solution, but it helps in being able
to use ours/theirs and long/short semantics with the data inside of
`Dlc`.

The hardest pill to swallow might be the changes to the tests in
`cfd.rs`, where a few test APIs have been modified to be able to pass
through the `Position`.

thomaseizinger reviewed (changes requested):
I think this is a good change but we cannot merge this because it breaks deserialization of old events and we haven't done https://github.com/itchysats/architecture/issues/3 yet.

Unless we figure out, how to make such changes we will need to band-aid the payout selection problem in a way that doesn't touch events.
